contracts/Plasma.sol

```diff
@@ -0,0 +1,211 @@
+pragma solidity 0.4.19;
+
+import 'zeppelin-solidity/contracts/ownership/Ownable.sol';
+import 'zeppelin-solidity/contracts/math/SafeMath.sol';
+import 'zeppelin-solidity/contracts/math/Math.sol';
+import './lib/MerkleProof.sol';
+import './lib/PriorityQueue.sol';
+import './Transaction.sol';
+
+import "./lib/RLP.sol";
+
+contract Plasma is Ownable {
+  mapping(address => bool) public operators;
+
+  modifier onlyOperator() {
+    if (operators[msg.sender]) _;
+  }
+
+  function setOperator(address _op, bool _status) public onlyOwner returns (bool success) {
+    operators[_op] = _status;
+    return true;
+  }
+
+  function isOperator() public view returns (bool success) {
+    return operators[msg.sender];
+  }
+
+  // blocks
+  struct Block {
+    bytes32 merkleRoot;
+    uint256 timestamp;
+  }
+
+  uint public blockCount;
+  mapping(uint => Block) public blocks;
+
+  event BlockSubmitted(uint indexed blockIndex, address operator, bytes32 indexed merkleRoot);
+
+  function submitBlock(bytes32 _merkleRoot, uint _blockIndex) public onlyOperator {
+    require(blockCount == _blockIndex);
```

Comment:
Not sure why `_blockIndex` has to be passed in. Seems that `blockIndex` is just `blockCount + 1`

Reply:
in case of race conditions (multiple operators submit blocks at the same time) the latter block gets canceled